packages/core/src/factory.ts

```diff
@@ -1,12 +1,15 @@
-import { configMergerFactory } from '.';
+import { configMergerFactory, PartialPrismConfig, PrismConfig } from '.';
 import { IPrism, IPrismComponents, IPrismConfig, IValidation } from './types';
 
 export function factory<Resource, Input, Output, Config, LoadOpts>(
+  defaultConfig: PrismConfig<Config, Input>,
```

Comment:
How about the opposite way? We could avoid passing `undefined` as first parameter.

```suggestion
  defaultComponents: Partial<IPrismComponents<Resource, Input, Output, Config, LoadOpts>>
  defaultConfig?: PrismConfig<Config, Input>,
```

Reply:
defaultConfig is mandatory and it is first arg. I don't see the point.